src/main/java/algorithm/CountingSortSnippet.java

```diff
@@ -0,0 +1,41 @@
+package algorithm;
+
+import java.util.Arrays;
+
+/*
+ * 30 Seconds of Java code library
+ *
+ */
+public class CountingSortSnippet {
+
+  /**
+   * Sort an array having zero or positive number with countingSort algorithm.
+   *
+   * @param arr array to sort
+   */
+  public static void countingSort(int[] arr) {
+    var max = Arrays.stream(arr).max().getAsInt();
+
+    var count = new int[max + 1];
+
+    for (var num : arr) {
+      count[num]++;
+    }
+
+    for (var i = 0; i <= max; i++) {
```

Comment:
@grzesiekkedzior Can you explain the problem here, please? I'm not getting it.

Reply:
When you run it then you get ArrayIndexOutOfBoundsException because of
```
... count[i - 1];
```
Start iteration from 1. You add array[N] to array[N - 1]. When N = 0 then N - 1 = array[-1].

Reply:
Okay. Thanks. I'll fix it.

Reply:
@grzesiekkedzior Fixed. Please check.